front/migrations/20241129_fix_intercom_conversation_parents.ts

```diff
@@ -0,0 +1,99 @@
+import { CoreAPI } from "@dust-tt/types";
+import assert from "assert";
+import _ from "lodash";
+import { QueryTypes, Sequelize } from "sequelize";
+
+import config from "@app/lib/api/config";
+import { DataSourceModel } from "@app/lib/resources/storage/models/data_source";
+import { makeScript } from "@app/scripts/helpers";
+
+const { CORE_DATABASE_URI } = process.env;
+const SELECT_CHUNK_SIZE = 1000;
+const UPDATE_CHUNK_SIZE = 10;
+
+makeScript({}, async ({ execute }, logger) => {
+  assert(CORE_DATABASE_URI, "CORE_DATABASE_URI is required");
+
+  const coreSequelize = new Sequelize(CORE_DATABASE_URI as string, {
+    logging: false,
+  });
+
+  const frontIntercomDataSources = await DataSourceModel.findAll({
+    where: { connectorProvider: "intercom" },
+  });
+  const coreAPI = new CoreAPI(config.getCoreAPIConfig(), logger);
+
+  for (const ds of frontIntercomDataSources) {
+    logger.info(`Processing data source ${ds.id}`);
+
+    let nextId = 0;
+    for (;;) {
+      const query = `
+          SELECT id, document_id, parents
+          FROM data_sources_documents
+          WHERE id > :nextId
+            AND EXISTS (SELECT 1
```

Comment:
as per irl, i'm not 100% but i think postgres will do:

- select docs
- for each line, execute the subselect
- limit by chunk size

which means, potentially, it will be quite expensive due to the subselect.

I would say join instead of just get rid of it and do all docs, regardless of the datasource.

Reply:
replaced the exist with a join, actually I'm going to keep the loop on the datasources to monitor more easily and because we loop over front's datasources